app.py

```diff
@@ -43,69 +54,65 @@ def profile():
     """
     The profile page of the app.
     """
+
+    if flask.request.method == "GET":
+        return render(data=current_user.profile.json)
+
     form = ProfileForm()
 
-    height_feet = form.height_feet.data
-    height_inches = form.height_inches.data
-
-    print(form.birthdate.data)
-    print(form.gender.data)
-    print(form.height_feet.data)
-    print(form.height_inches.data)
-    if height_feet is not None:
-        height = (int(height_feet) * 12) + int(height_inches)
-        print(height)
-
-    if form.validate_on_submit():
-        user_profile = Profile(
-            gender=form.gender.data,
-            height=height,
-            weight=form.weight.data,
-        )
-        db.session.add(user_profile)
-        db.session.commit()
-        print("User updated")
-        flash("Profile Updated!")
-        return redirect(url_for("profile"))
-
-    return render_template("profile.html", form=form)
-
-
-# NOTE user cannot be verified until DB is set up and conditional statements
-# will have to be updated
+    if not form.validate_on_submit():
+        return render(data=current_user.profile.json, form=form)
+
+    # delete existing profile (loop just in case)
+    for existing_profile in Profile.query.filter_by(user_id=current_user.id):
+        db.session.delete(existing_profile)
+
+    # update profile information
+    height_feet = form.height_feet.data or 0
+    height_inches = form.height_inches.data or 0
+
+    updated_profile = Profile(
+        user_id=current_user.id,
+        gender=form.gender.data,
+        height=height_feet * 12 + (height_inches % 12),  # mod to convert 12 to 0
+        weight=form.weight.data,
+        birth_year=form.birthdate.data.year,
+        birth_month=form.birthdate.data.month,
+        birth_day=form.birthdate.data.day,
+    )
+
+    db.session.add(updated_profile)
+    db.session.commit()
+
+    flash("Profile Updated!")
+    return redirect("profile")
+
+
 @app.route("/registration", methods=["GET", "POST"])
 def registration():
     """
     The registration page of the app. Allows new users to
     register for an account.
     """
+
+    if flask.request.method == "GET":
+        return render()
+
     form = RegisterForm()
 
-    # checks the input to see if the username and password are valid
-    if form.username.data is None:
-        flash("Please enter a username")
-    elif len(form.username.data) < 4:
-        flash("Please enter a username that has more than 4 characters")
-    elif len(form.username.data) > 20:
-        flash("Please enter a username that is less than 20 characters")
-    elif len(form.password.data) < 4:
-        flash("Please enter a password greater than 4 characters")
-    elif len(form.password.data) > 20:
-        flash("Please enter a password shorter than 20 characters")
```

Comment:
Are we adding new checks? We don't want the user to be able to input infinite length usernames.

Reply:
I moved the logic for all of the checks to the forms (See `forms.py`). This way, rather than seeing the message as a flash below the input, error messages are displayed natively on the individual input field with `reportValidity`.

Reply:
Oh sweet! That's perfect!